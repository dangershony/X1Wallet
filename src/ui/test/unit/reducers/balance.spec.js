import balanceReducer, { FETCH_BALANCE, FETCH_BALANCE_SUCCESS } from 'reducers/balance'

describe('reducers', () => {
  describe('balanceReducer', () => {
    it('should handle initial state', () => {
      expect(balanceReducer(undefined, {})).toMatchSnapshot()
    })

    it('should handle FETCH_BALANCE', () => {
      expect(balanceReducer(undefined, { type: FETCH_BALANCE })).toMatchSnapshot()
    })

    it('should handle FETCH_BALANCE_SUCCESS', () => {
      const walletBalance = {
        total_balance: 1,
        confirmed_balance: 1,
        unconfirmed_balance: 1,
      }
      const channelBalance = {
        balance: 1,
        pending_open_balance: 1,
      }
      expect(
        balanceReducer(undefined, { type: FETCH_BALANCE_SUCCESS, walletBalance, channelBalance })
      ).toMatchSnapshot()
    })

    it('should handle unknown action type', () => {
      expect(balanceReducer(undefined, { type: 'unknown' })).toMatchSnapshot()
    })
  })
})
