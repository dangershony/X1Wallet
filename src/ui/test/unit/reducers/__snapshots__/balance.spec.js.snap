// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers balanceReducer should handle FETCH_BALANCE 1`] = `
Object {
  "channelBalance": null,
  "fetchBalanceError": null,
  "isBalanceLoading": true,
  "walletBalance": null,
  "walletBalanceConfirmed": null,
  "walletBalanceUnconfirmed": null,
}
`;

exports[`reducers balanceReducer should handle FETCH_BALANCE_SUCCESS 1`] = `
Object {
  "channelBalance": 2,
  "channelBalanceConfirmed": 1,
  "channelBalancePending": 1,
  "fetchBalanceError": null,
  "isBalanceLoading": false,
  "walletBalance": 1,
  "walletBalanceConfirmed": 1,
  "walletBalanceUnconfirmed": 1,
}
`;

exports[`reducers balanceReducer should handle initial state 1`] = `
Object {
  "channelBalance": null,
  "fetchBalanceError": null,
  "isBalanceLoading": false,
  "walletBalance": null,
  "walletBalanceConfirmed": null,
  "walletBalanceUnconfirmed": null,
}
`;

exports[`reducers balanceReducer should handle unknown action type 1`] = `
Object {
  "channelBalance": null,
  "fetchBalanceError": null,
  "isBalanceLoading": false,
  "walletBalance": null,
  "walletBalanceConfirmed": null,
  "walletBalanceUnconfirmed": null,
}
`;
